### ISSUE TITLE
MS4 [Feature request]: create a vars file

### ISSUE THREAD

**Issue by parkervcp** [Griefed/ServerPackCreator#446]
### Whats the feature you would like to be added?

Export some sort of vars file that is just the vars in the start.sh that can be sourced when needed.

I would rather not need to run your shell script inside my own install script if I can help it.

### Anything else you would like to add?

I ask for this as the person who manages a large majority of pterodactyl game eggs.

This would make it easier to work with mod packs that are created using this tooling.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Griefed** [comment]
Hi parkervcp,

thanks for your feature request. 🙂
Are there any particular variables you would like to be exported to some sort of .env or vars-file?

> I would rather not need to run your shell script inside my own install script if I can help it.

I am curious about your reason(s) for this. Techincally, there is no difference between a regularly, handmade, server pack and a server pack made using ServerPackCreator. The files available inside the server pack should be, in the best case, identical. Server installations aren't shipped either, as CurseForge does not allow the server JARs to be included in the ZIP-archives.
Running foreign scripts is always risky, so I assume you have your own way of installing a server pack no matter who made it or how it was made.

Speaking from personal experience with using server packs: Many people, if not most, do not ship scripts of any kind with their pack at all.

Not having worked with Pterodactyl and having no knowledge about it I am wondering how the installation of a given server pack is affected by whether it was created with ServerPackCreator or not.

Should we not talk to each other in the next ~48 hours: Happy new year! 🎉

Cheers,
Griefed

**Comment by parkervcp** [comment]
I am really just looking for the following that are part of the install script you generate.

The pterodactyl install process just runs a script inside a docker container. It's a lot of functions to deal with many different modpack formats and layouts.

This is the script I use for curseforge https://gist.github.com/parkervcp/78ee503d347aceabf52cdafa21c91092#file-curseforge_installer-sh

```
MINECRAFT_VERSION="1.18.2"
MODLOADER="Forge"
MODLOADER_VERSION="40.1.86"
LEGACYFABRIC_INSTALLER_VERSION="0.11.1"
FABRIC_INSTALLER_VERSION="0.11.1"
QUILT_INSTALLER_VERSION="0.5.0"
MINECRAFT_SERVER_URL="https://launcher.mojang.com/v1/objects/c8f83c5655308435b3dcf03c06d9fe8740a77469/server.jar"
```

**Comment by Griefed** [comment]
This feature will be available in the next major release, that being version 4.x.x.
The file in question will be formatted like the following:

```properties
MINECRAFT_VERSION=1.12.2
MODLOADER=Forge
MODLOADER_VERSION=14.23.5.2860
LEGACYFABRIC_INSTALLER_VERSION=0.11.1
FABRIC_INSTALLER_VERSION=0.11.1
QUILT_INSTALLER_VERSION=0.5.0
MINECRAFT_SERVER_URL=https://launcher.mojang.com/v1/objects/886945bfb2b978778c3a0288fd7fab09d315b25f/server.jar
JAVAARGS="-Xms8G -Xmx8G -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=https://mcflags.emc.gs -Daikars.new.flags=true"
JAVA=java
```

and put into a separate file, probably something along the lines of `variables.txt`, not sure yet. 🙂 The first alpha of said major release will probably happen in early january.

Until then, take care and stay safe. 

Cheers,
Griefed

**Comment by Griefed** [comment]
First implementation available as of [4.0.0-alpha](https://github.com/Griefed/ServerPackCreator/releases/tag/4.0.0-alpha.12)

ServerPackCreator will generate a text-file called `variables.txt` inside the server pack now. :) 
Said file contains the default variables required to run a server via the default scripts generated and provided by SPC.

Example:

```
MINECRAFT_VERSION=1.12.2
MODLOADER=Forge
MODLOADER_VERSION=14.23.5.2860
LEGACYFABRIC_INSTALLER_VERSION=0.11.1
FABRIC_INSTALLER_VERSION=0.11.1
QUILT_INSTALLER_VERSION=0.5.0
MINECRAFT_SERVER_URL=https://launcher.mojang.com/v1/objects/886945bfb2b978778c3a0288fd7fab09d315b25f/server.jar
JAVA_ARGS="-Xms8G -Xmx8G -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=https://mcflags.emc.gs -Daikars.new.flags=true"
JAVA="java"
```
I will leave this issue open until the full release of SPC 4 in case any further conversation, discussion or feedback is required regarding this issue.

Cheers,
Griefed
